pkg/kubelet/cm/dra/manager.go

```diff
@@ -236,10 +236,21 @@ func (m *ManagerImpl) UnprepareResources(pod *v1.Pod) error {
 				claimInfo.cdiDevices, err)
 		}
 
+		// Delete last pod UID only if NodeUnprepareResource call succeeds.
+		// This ensures that status manager doesn't enter termination status
+		// for the pod. This logic is implemented in the m.PodCouldHaveUnpreparedResources
```

Comment:
Will need to fix this comment once we decide on a new name for `PodCouldHaveUnpreparedResources`

Reply:
done